kolibri/plugins/learn/assets/src/views/action-bar-search-box/index.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+
+  <div class="search-box-wrapper">
+    <div v-show="showSearchBox" :class="{ 'search-box-dropdown': showDropdownSearchBox }">
+      <search-box
+        ref="searchBox"
+        :icon="showDropdownSearchBox ? 'arrow_forward' : 'search'"
+        :width="showDropdownSearchBox ? 'calc(100vw - 120px)' : '150px'"
+      />
+    </div>
+
+    <ui-icon-button
+      v-show="showDropdownSearchBox"
+      icon="search"
+      type="primary"
+      color="primary"
+      @click="toggleDropdownSearchBox"
+    />
+
+    <div
+      v-show="showDropdownSearchBox && searchBoxOpen"
+      class="backdrop"
+      @click="toggleDropdownSearchBox">
+    </div>
+  </div>
+
+</template>
+
+
+<script>
+
+  import responsiveWindow from 'kolibri.coreVue.mixins.responsiveWindow';
+  import { PageNames } from '../../constants';
+  import uiIconButton from 'keen-ui/src/UiIconButton';
+  import searchBox from '../search-box';
+
+  export default {
+    $trNameSpace: 'learnSearchBox',
+    $trs: {
+      search: 'Search',
+      clear: 'Clear',
+    },
+    mixins: [responsiveWindow],
+    components: {
+      uiIconButton,
+      searchBox,
+    },
+    data() {
+      return {
+        searchQuery: this.searchTerm,
+        searchBoxOpen: false,
+      };
+    },
+    computed: {
+      showDropdownSearchBox() {
+        return this.windowSize.breakpoint <= 0;
+      },
+      showSearchBox() {
+        if (this.showDropdownSearchBox) {
+          return this.searchBoxOpen;
+        }
+        return true;
+      },
+    },
+    methods: {
+      toggleDropdownSearchBox() {
+        this.searchBoxOpen = !this.searchBoxOpen;
+        if (this.searchBoxOpen) {
+          this.$nextTick(() => {
+            this.$refs.searchBox.$refs.searchInput.focus();
+          });
+        }
+      },
+      search() {
+        if (this.searchQuery !== '') {
```

Comment:
Does anything happen if we search for an empty string? Looks like we already have checks for that in the searchbar component. Taking you to the search page might be expected behavior?